tests/test_utils/test_breakdown_validation.py

```diff
@@ -0,0 +1,49 @@
+from src.utils.breakdown_validation import breakdown_validation, run_breakdown_validation
+from pandas import DataFrame as pandasDF
+import pytest
+import logging
+
+class TestBreakdownValidation:
+    """Unit tests for breakdown_validation function."""
+
+    def create_input_df(self):
+        """Create an input dataframe for the test."""
+        input_cols = ["reference", "202", "203", "222", "223", "204","205", "206", "207","219","220", "209", 
+        "221", "210", "211", '212','214','216','242','250','243','244','245','246','247','248', '249', '218',
+        '225','226','227','228','229','237','302', '303','304','305','501','503','505','507','502','504','506',
+        '508','405', '407','409', '411', '406', '408', '410', '412', "871"]
+
+        data = [
+             ['A', 1, 30, 15, 15, 40, 10, 10, 20, 10, 15, 20, 10, 45, 85, 10, 10, 10, 10, 10, 10, 10, 10, 1, 1, 2, 1, 85, 20, 20, 20, 20, 2, 3, 50, 20, 10, 80, 50, 20, 25, 95, 60, 20, 10, 90, 80, 20, 10, 110, 80, 90, 20, 190, None],  
+             ['B', 10, 30, 15, 15, 40, 10, 10, 20, 10, 15, 20, 10, 45, 85, 10, 10, 10, 10, 10, 10, 10, 10, 1, 1, 2, 1, 85, 20, 20, 20, 20, 2, 3, 50, 20, 10, 80, 50, 20, 25, 95, 60, 20, 10, 90, 80, 20, 10, 110, 80, 90, 20, 190, None],
+             ]
```

Comment:
Please can you add additional references - one with all nulls so we can test it is excluded, and one with a 0 in the total and nulls in the breakdown, so we can test that the new functionality works